packages/j/juce/xmake.lua

```diff
@@ -0,0 +1,237 @@
+        add_urls("https://github.com/juce-framework/JUCE/releases/download/7.0.11/juce-7.0.11-osx.zip")
+        add_versions("7.0.11", "45c3bdd8d7f00d5d990cc1f85066d49629d40b411ae9624bda44524cc5a69058")
+    end
+
+    local modules = {
+        juce_analytics = {
+            default = false,
+            readonly = false,
+            deps = {"juce_gui_basics"}
+        },
+        juce_audio_basics = {
+            default = false,
+            readonly = false,
+            links = {
+                iphoneos = {"Accelerate"},
+                macosx = {"Accelerate"},
+            },
+            deps = {"juce_core"}
+        },
+        juce_audio_devices = {
+            default = false,
+            readonly = false,
+            links = {
+                iphoneos = {"CoreAudio", "CoreMIDI", "AudioToolbox"},
+                macosx = {"CoreAudio", "CoreMIDI", "AudioToolbox", "AVFoundation"},
+                linux = {"alsa"},
+                mingw = {"winmm"}
+            },
+            deps = {"juce_audio_basics", "juce_events"}
+        },
+        juce_audio_formats = {
+            default = false,
+            readonly = false,
+            links = {
+                iphoneos = {"AudioToolbox", "QuartzCore"},
+                macosx = {"CoreAudio", "CoreMIDI", "QuartzCore", "AudioToolbox"},
+            },
+            deps = {"juce_audio_basics", "juce_events"}
+        },
+        juce_audio_plugin_client = {
+            default = false,
+            readonly = false,
+            deps = {"juce_audio_processors"}
+        },
+        juce_audio_processors = {
+            default = false,
+            readonly = false,
+            links = {
+                iphoneos = {"AudioToolbox"},
+                macosx = {"CoreAudio", "CoreMIDI", "AudioToolbox"},
+            },
+            deps = {"juce_gui_extra", "juce_audio_basics"}
+        },
+        juce_audio_utils = {
+            default = false,
+            readonly = false,
+            links = {
+                iphoneos = {"CoreAudioKit"},
+                macosx = {"CoreAudioKit", "DiscRecording"},
+            },
+            deps = {"juce_audio_processors", "juce_audio_formats", "juce_audio_devices"}
+        },
+        juce_box2d = {
+            default = false,
+            readonly = false,
+            deps = {"juce_graphics"}
+        },
+        juce_core = {
+            default = true,
+            readonly = true,
+            links = {
+                windows = {"kernel32", "user32", "shell32", "gdi32", "vfw32", "comdlg32", "winmm", "wininet", "rpcrt4", "ole32", "advapi32", "ws2_32", "Version", "Imm32", "Shlwapi"},
+                linux = {"rt", "dl", "pthread"},
+                macosx = {"Cocoa", "Foundation", "IOKit", "Security"},
+                iphoneos = {"Foundation"},
+                mingw = {"uuid", "wsock32", "wininet", "version", "ole32", "ws2_32", "oleaut32", "imm32", "comdlg32", "shlwapi", "rpcrt4", "winmm"}
+            }
+        },

```

Comment:
Please use add_configs to configure them, one per line, to ensure a consistent style of repository packages.